### ISSUE TITLE
New code execution and read, stdin

### ISSUE THREAD

**Issue by ChrHorn** [julia-vscode/julia-vscode#785]
There seems to be an issue with the new terminal communication and reading stdin. Executing the file
```julia
for n in 1:5
    println(n)
    readline()
end
```
through vscode will drop back to the REPL prompt after the first iteration. Loading the same file with include() works as expected.

Version: 0.12.0-beta.2

**Comment by davidanthoff** [comment]
Hm, that is a tricky one... We are executing that piece of code on an async task, and it seems that using `readline` doesn't play well when used from a background task? @pfitzseb, have you come across this issue in Juno? I would think you might have the same issue?

**Comment by pfitzseb** [comment]
We do have [that same issue](https://github.com/JunoLab/Juno.jl/issues/118), yes.

As a user you can kinda work around this by typing a character in the REPL after the `readline` prompt, pressing enter, and then typing what you want `readline` to catch and press enter again.

Fixing this on the IDE end has turned out to be really difficult with the current REPL implementation. We might be able to push the code we want to execute to the REPL backend directly, but IIRC you can't actually get the relevant `Channel` to do that. 
An alternative to that would be to make some changes to the REPL code (kinda like what we're doing already by overwriting `on_done`) to disregard anything happening on `stdin` when something is evaluated. Couldn't figure out how to do that as of yet though...

**Comment by davidanthoff** [comment]
Thanks for those insights, @pfitzseb! I'm putting this on the backlog, we won't be able to solve this in v0.12.0.

**Comment by laborg** [comment]
Is there a recommended workaround?

**Comment by pfitzseb** [comment]
Yes. Don't use inline evaluation for code that takes input from stdin :)

**Comment by daler6** [comment]
Are there any plans to address this issue? It is rather annoying.

**Comment by daler6** [comment]
Also, I created a bit of a workaround for using readline() in a .jl file, but not for a Jupyter notebook. Any thoughts on that?

**Comment by DSGeoff** [comment]
I agree that this is rather annoying, but I think it's more than that.  Looking above, it appears this issue has been duplicated many times and has been around for more than 4 years.  This is a pretty standard thing where you need to read from a database but putting your password in your code is a big no-no.  A simple way is to prompt the user for their password directly within the connection.  Does anyone have a good solution for this that can be used within VS Code?

**Comment by DSGeoff** [comment]
The solution suggested above is to just not use the feature.  I would like to suggest better solutions that seem to work in some cases, though it's possible these could depend on versions of various software involved.  These are in order of my preference:

1. Use another editor, such as Jupyter notebook (not inside of VS Code) where this works as expected
2. Within VS Code
a. Run code in the VS Code editor (not in the repl).  When the prompt comes up, enter *something* other than blank and hit enter, then enter what you really want to enter.  The prompt ends up with the second value.  If you don’t input anything and just hit enter at first, it doesn’t seem to advance things to where the next input works.  You’d still have to enter something other than blank, then what you really want for it to get accepted.
b. Copy the full code into the repl and run it there.  There was a comment on one issue that said, “Ctrl-Enter will also do the trick (because it makes the pasta for you ;))”, but in my experience that doesn’t change anything.  It’s possible there’s additional assumed knowledge that I don’t have that makes it work.

**Comment by pfitzseb** [comment]
2b worked at the time the original comment was written, where that shortcut would in fact copy-paste the current line into the REPL without using inline evaluation.

The suggested workaround right now is to paste the entrypoint into the REPL and run it there manually.
